google/cloud/bigtable/internal/query_plan.cc

```diff
@@ -21,11 +21,30 @@ namespace cloud {
 namespace bigtable_internal {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 
+bool QueryPlan::IsExpired() const {
+  auto total_duration =
+      std::chrono::seconds(response_.valid_until().seconds()) +
+      std::chrono::nanoseconds(response_.valid_until().nanos());
+
+  auto valid_until = std::chrono::system_clock::time_point() + total_duration;
+  return valid_until <= std::chrono::system_clock::now();
+}
+
 std::string const& QueryPlan::prepared_query() const {
+  std::lock_guard<std::mutex> lock(mu_);
+  if (IsExpired()) {
+    static std::string const kEmpty;
+    return kEmpty;
+  }
   return response_.prepared_query();
 }
 
 google::bigtable::v2::ResultSetMetadata const& QueryPlan::metadata() const {
```

Comment:
For now let's change the return type to `StatusOr<ResultSetMetadata>` and return `kUnavailable` when IsExpired.

Reply:
Same as above, done.